sqlglot/optimizer/annotate_types.py

```diff
@@ -421,3 +484,14 @@ def _annotate_by_args(
             )
 
         return expression
+
+    def _annotate_dateadd(self, expression: exp.IntervalOp) -> exp.IntervalOp:
+        self._annotate_args(expression)
+
+        if expression.this.type.is_type(*exp.DataType.TEXT_TYPES):
```

Comment:
Should we instead phrase this as `expression.this.type in exp.DataType.TEXT_TYPES` to avoid unpacking?

Reply:
ah sure